src/libs/actions/IOU.js

```diff
@@ -722,11 +723,219 @@ function payIOUReport({
     return promiseWithHandlers;
 }
 
+/**
+ * @param {String} chatReportID - Report ID of the chat where the IOU is.
+ * @param {Number} amount - IOU amount in cents.
+ * @param {String} currency - IOU currency.
+ * @param {String} comment
+ * @param {Array} participants - Participants on the IOU
+ */
+function sendMoneyWithWallet(chatReportID, amount, currency, comment, participants) {
+    const chatReport = _.empty(chatReportID) ? chatReports[`${ONYXKEYS.COLLECTION.REPORT}${chatReportID}`] : ReportUtils.buildOptimisticChatReport(participants);
+    const ownerEmail = participants[0].login;
```

Comment:
Will participants 0 _always_ be the owner?

Reply:
I think this is the way it's organised, but let me dig a bit deeper

Reply:
it's expected for the `0` to be the owner of the report, I don't know if I can garantee that, but this seems bigger than this PR. 
I aggree that maybe some further logic is needed for the participants. 
We seem to expect in several places that this always like that, but I'm not sure. I think the logic comes from [here](https://github.com/Expensify/App/blob/e65107b36f4266c477e57049e06af4f43e022fb4/src/libs/actions/Report.js#L67-L80)
But I only see the owner email being added to the beggining if the report is a PolicyExpenseChat, not sure if this the case.

If the chat report already exists, and the other user created it, and you send them money I'm not sure this would be true (the `0` being the owner)...
let me do some tests

Reply:
This should be fine and correct. It is also used in other of the IOU commands, but I agree this is not clear and we might want to refactor this for better DX and also minimizing chances of errors going forwards.

Reply:
I think we will have merge conflicts afterwards, this is how I'm thinking of making this more clear and would like your feedback. I think Andre and I need to settle on the same solution so that code is consistent. The recipient/participant has to be an object here 

<img width="650" alt="Screenshot 2022-11-29 at 18 31 45" src="https://user-images.githubusercontent.com/9680864/204601022-fcc9e857-cd9c-4004-b2bc-d6ced57ac827.png">

Reply:
not sure this is the correct file for your comment @youssef-lr 

@stitesExpensify and @vitHoracek I agree that this might need some refactoring, but maybe not as part of this PR that is already big.
Maybe we can propose something on WhatsNext

For now I think I'm going to create the ReportUtils Method and then find the first email from the participants that is not the owner to be the other email

Reply:
I dont think we would need WN process for this, we could just discuss this in engineering chat with people who know the IOU flow.

Reply:
> @sketchydroide I was suggesting we send the participant (the user we select from the OptionsList) to the SendMoney commands, instead of extracting it there from the array. What do you think?

I think that might work, but we still need the owner, I think that one could/should be the one that is currently logged in

Reply:
so I might change the logic I implemented earlier